mipengine/controller/node_registry.py

```diff
@@ -1,236 +1,38 @@
-
-        Parameters
-        ----------
-        node_id: the id of the node
-        data_model: the data model of the datasets
-        wanted_datasets: the datasets to look for
-
-        Returns
-        -------
-        some, all or none of the wanted_datasets that are located in the node
-        """
+        }
 
-        for node in self.nodes:
-            if node.id == node_id:
-                if data_model not in node.datasets_per_data_model.keys():
-                    raise ValueError(
-                        f"Data model '{data_model}' is not available in the node '{node_id}'."
-                    )
-                node_datasets = node.datasets_per_data_model[data_model]
-                break
-        else:
-            raise ValueError(
-                f"Node '{node_id}' could not be found in the node registry."
-            )
-
-        return list(set(node_datasets).intersection(wanted_datasets))
-
-    # returns a list of all the currently available data_models on the system
-    # without duplicates
-    def get_all_available_data_models(self) -> List[str]:
-        all_local_nodes = self.get_all_local_nodes()
-        tmp = [local_node.datasets_per_data_model for local_node in all_local_nodes]
-        all_existing_data_model = set().union(*tmp)
-        return list(all_existing_data_model)
-
-    # returns a dictionary with all the currently available data_models on the
-    # system as keys and lists of datasets as values. Without duplicates
-    def get_all_available_datasets_per_data_model(self) -> Dict[str, List[str]]:
-        all_local_nodes = self.get_all_local_nodes()
-        tmp = [node_info.datasets_per_data_model for node_info in all_local_nodes]
-        from collections import defaultdict
-        from itertools import chain
-        from operator import methodcaller
-
-        dd = defaultdict(list)
-        dict_items = map(methodcaller("items"), tmp)
-        for k, v in chain.from_iterable(dict_items):
-            dd[k].extend(v)
-
-        return dict(dd)
-
-    def data_model_exists(self, data_model: str):
-        for node_info in self.get_all_local_nodes():
-            if not node_info.datasets_per_data_model:
-                continue
-            if data_model in node_info.datasets_per_data_model.keys():
-                return True
-        return False
-
-    def dataset_exists(self, data_model: str, dataset: str):
-        for node_info in self.get_all_local_nodes():
-            if not node_info.datasets_per_data_model:
-                continue
-            if data_model not in node_info.datasets_per_data_model:
-                continue
-            if not _have_common_elements(
-                node_info.datasets_per_data_model[data_model], [dataset]
-            ):
-                continue
-            return True
-        return False
+    def get_global_node(self) -> NodeInfo:
+        return self.nodes["globalnode"]
 
+    def get_all_local_nodes(self) -> Dict[str, NodeInfo]:
```

Comment:
I changed the nodes in general to dictionaries because the id is unique and it will be easier to iterate and find the proper NodeInfo that you will need.
With list to find a NodeInfo of a specific node you should: 
`nodes  = [node for node in self.nodes if node.id == "somenodeid"]
#Then cause we knew that it was unique we had to pick the first.
node = nodes[0]
`
With the dict to find a specific NodeInfo you just need to provide the key and you got the proper NodeInfo:
`node = self.nodes["somenodeid"]`